src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double firstBase;
+    private double secondBase;
+    private double height;
+
+    public IsoscelesTrapezoid(String color, double firstBase, double secondBase, double height) {
```

Comment:
Instead of generating values like firstBase, secondBase, and height in the constructor, consider passing them as parameters from a 'FigureSupplier' class to adhere to the single responsibility principle and to make the class more testable.